ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py

```diff
@@ -84,11 +84,15 @@ def __init__(self, name: str, connection: ConnectionBase, tags: list | None = No
 
         super().__init__(name, tags, disable_cache=False)
         self.check_mode = check_mode
-        # In check_mode we don't care that we cannot connect to the device
-        if self.check_mode or (plugin_name := connection._sub_plugin.get("name")) == ANSIBLE_EOS_PLUGIN_NAME:
-            self._connection = connection
-        else:
-            raise AristaAvdError(message=f"The provided Ansible connection does not use EOS HttpApi plugin: {plugin_name}")
+
+        try:
+            # In check_mode we don't care that we cannot connect to the device
+            if self.check_mode or (plugin_name := connection._sub_plugin.get("name")) == ANSIBLE_EOS_PLUGIN_NAME:
+                self._connection = connection
+            else:
+                raise AristaAvdError(message=f"The provided Ansible connection does not use EOS HttpApi plugin: {plugin_name}")
+        except AttributeError as err:
+            raise AristaAvdError(message="AVD could not determine ansible connection. Please configure ansible_connection as `httpapi` for this host.") from err
```

Comment:
```suggestion
            raise AristaAvdError(message="AVD could not determine the Ansible connection plugin used. Please ensure that the 'ansible_network_os' and 'ansible_connection' variables are set to 'eos' and 'httpapi' respectively for this host.") from err
```

Reply:
Fixed.
If ansible_network_os is not defined and ansible_connection defined as httpapi then we face below error
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'Connection' object has no attribute 'nonetype'
fatal: [leaf2-dc1]: FAILED! => {"msg": "Unexpected failure during module execution: 'Connection' object has no attribute 'nonetype'", "stdout": ""}
```
and this is from
```
File "/home/maheshkumar/Projecct/ansible-avd/ansible_collections/ansible/netcommon/plugins/plugin_utils/connection_base.py", line 50, in __getattr__
    raise AttributeError(
AttributeError: 'Connection' object has no attribute 'nonetype'
fatal: [leaf2-dc1]: FAILED! => {
    "msg": "Unexpected failure during module execution: 'Connection' object has no attribute 'nonetype'",
    "stdout": ""
}
cc: @gmuloc 
```

Reply:
done